### PR TITLE
Create page 404 as React component, use it for listings

### LINKED ISSUE #2380: Show 404 page for non-existent listings
Use https://www.originprotocol.com/404 for background image

### PR DESCRIPTION
**Pull request by zharinov**
Closes #2380

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: dapps/marketplace/src/components/Error404.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+
+const Error404 = props => {
+  return (
+    <div className="error-404">
+      <div className="container">
+        <div className="row">
```
Comment (nick) [inline dapps/marketplace/src/components/Error404.js]: I think we can remove the `row` and `col-12` and put the `props.children` directly in the container

File: dapps/marketplace/src/pages/listing/Listing.js [hunk 2]
```diff
@@ -33,23 +34,30 @@ class Listing extends Component {
             </fbt>
           }
         />
-        <Query query={query} variables={vars}>
+        <Query query={query} variables={vars} errorPolicy="all">
           {({ networkStatus, error, data, refetch }) => {
+            const error404 = (
```
Comment (nick) [inline dapps/marketplace/src/pages/listing/Listing.js]: Can we make this a component put at the top of the file? (ie outside the render method)

File: dapps/marketplace/src/components/Error404.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          top: 50%
+          width: 30%
+  
+        .dude
+          position: absolute
+          left: 39%
+          top: 36%
+          width: 8%
+    
+  @media (max-width: 768px)
```
Comment (nick) [inline dapps/marketplace/src/components/Error404.js]: We've been using `@media (max-width: 767.98px)` to be consistent with bootstrap

**Comment by micahalcorn** [comment]
How about giving `error-404` a background color of `#f7f9fb` ❔

**Comment by zharinov** [comment]
> How about giving `error-404` a background color of `#f7f9fb` ❔

Thanks, I felt something is wrong, but didn't notice background color

**Comment by micahalcorn** [comment]
Don't forget to run the translation command. 🌐

**Comment by zharinov** [comment]
I did run `npm run translate` and it resulted to empty-object JSONs at `public/translations` as well as `crowdin/all-messages.json`. Since I used existing i18n string in this PR, I suppose that's correct output, isn't it? Should it be commited anyway to meet the i18n pipeline?

**Comment by nick** [comment]
Thank you for your work here @zharinov!

**Comment by zharinov** [comment]
Thanks @nick for your comments 👍

**Pull request merged**